plutus-metatheory/src/Algorithmic/CEK.lagda.md

```diff
@@ -72,43 +68,47 @@ data Value : (A : ∅ ⊢Nf⋆ *) → Set where
    → Value (nf (embNf A · ƛ (μ (embNf (weakenNf A)) (` Z)) · embNf B))
    → Value (μ A B)
 
-  V-con : {tcn : TyCon _}
-    → (cn : TermCon {∅} (con tcn))
-    → Value (con tcn)
+  V-con : ∀{A : ∅ ⊢Nf⋆ ♯}
+    → ⟦ A ⟧ 
+    → Value (con A)
```

Comment:
Why is that? The `TermCon` that we would have here would we a dependent pair of a type and its meaning, so it would be equivalent but with one more level of indirection.

Reply:
> The `TermCon` that we would have here would we a dependent pair of a type and its meaning

I want the implementation to have a `TermCon` that isn't just (morally) a dependent pair, but instead is a bunch of constructors with type tags in their names rather than in a separate field. That is faster than storing tags separately and isn't much more boilerplate (we'd still have to keep type tags around for fast unlifting of lists and pairs).

Reply:
OK. But you are talking about the untyped case. For the typed case, this version is potentially more efficient, because you only can get something of the correct type, so you wouldn't need to check for tags. Anyway, efficiency is not currently an issue for the formalisation, as long as its performance is not unreasonable and can be used for testing.

Reply:
> Anyway, efficiency is not currently an issue for the formalisation

Sure, I'm just letting you know what is planned for the implementation, you don't need to follow that in the formalization, but I thought you might want to know.